tsdb/chunkenc/histo.go

```diff
@@ -350,6 +361,12 @@ func (a *HistoAppender) AppendHistogram(t int64, h histogram.SparseHistogram) {
 	var tDelta, cntDelta, zcntDelta int64
 	num := binary.BigEndian.Uint16(a.b.bytes())
 
+	if value.IsStaleNaN(h.Sum) {
+		// Emptying out other fields to write empty meta in case of
+		// first histogram in the chunk.
+		h = histogram.SparseHistogram{Sum: h.Sum}
+	}
```

Comment:
I think the only code path that gives us a stale marker here is already ensuring that the histogram is otherwise empty. So we don't need this, do we?

Reply:
I plan to remove the cleanup in appendHistogram() in head.go and have it here since this chunk need not be used just by that code path. The case of adding stale via Append() is scraping specific.